app/src/main/java/com/github/swent/echo/compose/event/EventScreen.kt

```diff
@@ -1,52 +1,100 @@
 import androidx.compose.foundation.gestures.detectTapGestures
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.rememberScrollState
 import androidx.compose.foundation.verticalScroll
 import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.LocalFocusManager
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.unit.dp
 import com.github.swent.echo.R
+import com.github.swent.echo.viewmodels.event.EventStatus
 import com.github.swent.echo.viewmodels.event.EventViewModel
+import kotlinx.coroutines.launch
 
 val EVENT_PADDING_BETWEEN_INPUTS = 10.dp
 /** This screen allows the user to create or edit an event. */
+@SuppressLint("CoroutineCreationDuringComposition")
 @Composable
 fun EventScreen(
     title: String,
-    onEventSaveButtonPressed: () -> Unit,
+    onEventSaved: () -> Unit,
     onEventBackButtonPressed: () -> Unit,
     eventViewModel: EventViewModel
 ) {
     val focusManager = LocalFocusManager.current
-    Column(
-        modifier =
-            Modifier.verticalScroll(rememberScrollState()).pointerInput(Unit) {
-                detectTapGestures(onTap = { focusManager.clearFocus() })
-            }
-    ) {
-        EventTitleAndBackButton(title = title, onBackButtonPressed = onEventBackButtonPressed)
-        // all the inputs for an event
-        EventPropertiesFields(eventViewModel = eventViewModel)
-        // save button
-        OutlinedButton(
+    val scope = rememberCoroutineScope()
+    val snackBarHostState by remember { mutableStateOf(SnackbarHostState()) }
+    val eventStatus = eventViewModel.status.collectAsState()
+
+    var saveButtonText by remember { mutableIntStateOf(R.string.edit_event_screen_save) }
+    var saveButtonClicked by remember { mutableStateOf(false) }
+    val localContext = LocalContext.current
+
+    if (saveButtonClicked && eventStatus.value is EventStatus.Saved) {
+        saveButtonClicked = false
+        saveButtonText = R.string.edit_event_screen_save
+        onEventSaved()
+    } else if (eventStatus.value is EventStatus.Error) {
+        scope.launch {
```

Comment:
Use `LaunchEffect(eventStatus.value) {....}`

Reply:
Thanks I wasn't aware of this function